shop/src/main/java/ro/msg/learning/shop/controller/ProductController.java

```diff
@@ -0,0 +1,40 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.dto.ProductDto;
+import ro.msg.learning.shop.exception.LocationNotFoundException;
+import ro.msg.learning.shop.exception.ProductCategoryNotFoundException;
+import ro.msg.learning.shop.exception.ProductNotFoundException;
+import ro.msg.learning.shop.service.ProductService;
+
+import java.util.List;
+import java.util.UUID;
+
+@RequestMapping("/product")
+@Validated
+@RestController
+public class ProductController {
+    @Autowired
+   private ProductService productService;
+
+    @PostMapping("/new")
```

Comment:
So you have a `@RequestMapping("/product")` for the whole Controller. Then you have this POST method with additional path "/new". You don't need this. There are some best practices we should follow.
Reference: https://www.freecodecamp.org/news/rest-api-best-practices-rest-endpoint-design-examples/